### PR TITLE
Fetch dish items dynamically

### PR DESCRIPTION
**Pull request by kprakhar04**
Dish items are getting loaded dynamically
Screenshot 2022-03-09 at 17 55 47

**Deploy Preview for this PR -> https://deploy-preview-3--haul-meal.netlify.app/**

### CODE REVIEW

**Comment by mahendarsingh1** [comment]
in DishMenu.js, inside the map function of currItem you can send the item object as props instead of sending all the properties of it separately by assigning it to new object. this could reduce some lines of code in it. and make it more simpler

**Comment by kprakhar04** [comment]
> in DishMenu.js, inside the map function of currItem you can send the item object as props instead of sending all the properties of it separately by assigning it to new object. this could reduce some lines of code in it. and make it more simpler


Ya you're right, done!

File: src/Components/DishItems/DishMenu.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const rows = [];
+  Object.keys(items).forEach((category) => {
+    const currItem = items[category];
+    rows.push(
+      <DishCategoryRow
+        key={category}
+        heading={category}
+        subtitle={currItem.length}
+      >
+        {currItem.map((item) => {
```
Comment (jamruthsai) [inline src/Components/DishItems/DishMenu.js]: Can make separate fn/component

File: src/Components/DishItems/DishMenu.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import DishCategoryRow from "./DishCategoryRow";
+import "./DishMenu.css";
+import DishRow from "./DishRow";
+
+const getDishTemplate = function (items) {
```
Comment (jamruthsai) [inline src/Components/DishItems/DishMenu.js]: Can make this separate component

File: src/utility/helpers.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { CURRENCYNAMETOSYMBOL } from "../constants/currency";
+
+const ICON_ROOT_DIR = "/assets/icons";
```
Comment (jamruthsai) [inline src/utility/helpers.js]: Should be in constant folder

File: src/Components/DishItems/DishMenu.js (hunk 1 above)
Comment (kprakhar04) [reply src/Components/DishItems/DishMenu.js]: It's a separate component, it's getting rendered as its children

File: src/utility/helpers.js (hunk 3 above)
Comment (kprakhar04) [reply src/utility/helpers.js]: yes

File: public/data/dishes.json [hunk 4]
```diff
@@ -0,0 +1,84 @@
+{
```
Comment (jamruthsai) [inline public/data/dishes.json]: You can delete this file

File: src/components/DishItems/DishItems.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    if (isLoading) {
+      return <Spinner text="Loading..." />;
+    }
+
+    if (error) {
+      const { message, statusCode } = error;
+      return <ErrorHandler message={message} statusCode={statusCode} />;
+    }
+
+    const sidebarLinks = _keys(dishItems);
```
Comment (jamruthsai) [inline src/components/DishItems/DishItems.js]: define at top

File: src/components/DishItems/Dishes/Menu/Menu.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+      heading={category}
+      subtitle={currItem.length}
+      items={currItem}
+      renderChild={getMenuRowRenderProps}
+    />
+  );
+};
+
+const renderMenu = (items) => {
+  return _map(_keys(items), (itemCategory) =>
```
Comment (jamruthsai) [inline src/components/DishItems/Dishes/Menu/Menu.js]: inline function

File: src/commonComponents/loader/index.js [hunk 7]
```diff
@@ -1,7 +1,15 @@
 import Spinner from "./Spinner";
 
+<<<<<<< HEAD
 const Loader = {
```
Comment (jamruthsai) [inline src/commonComponents/loader/index.js]: Resolve this merge conflict

File: src/commonComponents/primaryButton/PrimaryButton.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+// prop-types
+import Proptypes from "prop-types";
+
+// lodash
+import _noop from "lodash/noop";
+
+// css
+import "./primaryButton.css";
+
+const PrimaryButton = ({ btnText, handleClick, className }) => {
```
Comment (jamruthsai) [inline src/commonComponents/primaryButton/PrimaryButton.js]: Instead take props and desturcture

File: src/App.js [hunk 9]
```diff
@@ -1,6 +1,6 @@
-// component
+// components
 import Header from "./components/header";
-import SubHeader from "./components/subHeader";
+import RestaurantWrapper from "./components/restaurantWrapper";
```
Comment (anujtekion) [inline src/App.js:3]: we can call this RestaurantPage

File: src/components/restaurantWrapper/RestaurantWrapper.js [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+
+class RestaurantWrapper extends React.Component {
+  state = INITIAL_STATE;
+
+  componentDidMount() {
+    this.fetchRestaurant();
+  }
+
+  fetchRestaurant() {
+    getRestaurantDetails(DEFAULT_RESTAURANT_ID)
```
Comment (anujtekion) [inline src/components/restaurantWrapper/RestaurantWrapper.js:30]: use promise.all

File: src/components/restaurantWrapper/RestaurantWrapper.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
+// react
```
Comment (anujtekion) [inline src/components/restaurantWrapper/RestaurantWrapper.js:1]: restaurant.js

File: src/components/subHeader/SubHeader.js [hunk 12]
```diff
@@ -1,132 +1,137 @@
...
+        title={ratingValue.toFixed(1)}
+        subtitle={`${ratingCount}+ ratings`}
+      />
+      <Placeholder title={`${deliveryTime} mins`} subtitle="Delivery Time" />
+      <Placeholder title={cost} subtitle={`Cost for ${count}`} />
+    </>
+  );
+};
+
+const getBreadCrumbConfig = defaultMemoize(breadCrumbConfig);
```
Comment (anujtekion) [inline src/components/subHeader/SubHeader.js:47]: usecase of useMemo hook

File: src/components/subHeader/SubHeader.js [hunk 13]
```diff
@@ -1,132 +1,137 @@
...
+  ],
+  ratingCount: 20,
+  ratingValue: 4.1,
+  deliveryTime: 40,
+  cost: 500,
+  count: "two",
+};
+
+SubHeader.propTypes = {
+  restaurantDetails: PropTypes.shape({
```
Comment (anujtekion) [inline src/components/subHeader/SubHeader.js:116]: check proptypes

File: src/components/subHeader/SubHeader.js [hunk 14]
```diff
@@ -1,132 +1,137 @@
...
-    const thumbnail = restaurantReader.thumbnail(restaurantDetails);
-    const name = restaurantReader.name(restaurantDetails);
-    const category = restaurantReader.category(restaurantDetails);
-    const street = restaurantReader.street(restaurantDetails);
-    const locality = restaurantReader.locality(restaurantDetails);
-    const city = restaurantReader.city(restaurantDetails);
-    const offers = restaurantReader.offers(restaurantDetails);
 
-    const crumbs = this.getBreadCrumbConfig(city, locality, name);
+const renderPlaceholder = ({
```
Comment (anujtekion) [inline src/components/subHeader/SubHeader.js:26]: do little bit digging why this should be outside/inside

File: src/components/subHeader/components/header/Header.js [hunk 15]
```diff
@@ -4,11 +4,22 @@ import PropTypes from "prop-types";
 //lodash
 import _noop from "lodash/noop";
 
+// icons
+// icon
```
Comment (anujtekion) [inline src/components/subHeader/components/header/Header.js:8]: remove

File: src/components/subHeader/SubHeader.js (hunk 14 above)
Comment (anujtekion) [inline src/components/subHeader/SubHeader.js:26]: read about useCallback

File: src/utility/helpers.js [hunk 16]
```diff
@@ -1,6 +1,27 @@
+// helpers
```
Comment (anujtekion) [inline src/utility/helpers.js]: Rules about writing utils

1. Every function that you export should have separate file
2. file name will be the function name that you are exporting
3. No helper naming inside util folder

File: src/constants/directoryEndPoint.js [hunk 17]
```diff
@@ -0,0 +1 @@
+export const ICON_ROOT_DIR = "/assets/icons";
```
Comment (anujtekion) [inline src/constants/directoryEndPoint.js]: should be inside where the function is written

File: src/constants/currency.js [hunk 18]
```diff
@@ -0,0 +1,5 @@
+export const CURRENCYNAMETOSYMBOL = {
```
Comment (anujtekion) [inline src/constants/currency.js]: CURRENCY_NAME_TO_SYMBOL

File: src/components/dishItems/DishItems.js [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+// css
+import "./dishItems.css";
+
+const DishItems = (props) => {
+  const { dishItems } = props;
+
+  const sidebarLinks = _keys(dishItems);
+
+  return (
+    <main className="dish flex">
```
Comment (anujtekion) [inline src/components/dishItems/DishItems.js]: remove flex

File: src/commonComponents/sidebar/Sidebar.js [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+  handleHashChange = () => {
+    this.setState({
+      hashChanged: true,
+    });
+  };
+
+  renderLink = (label, index) => {
+    const { hashChanged } = this.state;
+    const url = generateUrl(label);
+    let isEqual = checkEqualityWithHash(url);
```
Comment (anujtekion) [inline src/commonComponents/sidebar/Sidebar.js:43]: rename

File: src/commonComponents/sidebar/helpers/sidebar.helpers.js [hunk 21]
```diff
@@ -0,0 +1,17 @@
+// lodash
```
Comment (anujtekion) [inline src/commonComponents/sidebar/helpers/sidebar.helpers.js]: .general.js, never say .helpers inside helpers

File: src/commonComponents/sidebar/Sidebar.js [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+
+  handleHashChange = () => {
+    this.setState({
+      hashChanged: true,
+    });
+  };
+
+  renderLink = (label, index) => {
+    const { hashChanged } = this.state;
+    const url = generateUrl(label);
```
Comment (anujtekion) [inline src/commonComponents/sidebar/Sidebar.js]: rename

File: src/commonComponents/sidebar/Sidebar.js [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+
+class Sidebar extends React.Component {
+  state = INITIAL_STATE;
+
+  componentDidMount() {
+    window.addEventListener("hashchange", this.handleHashChange);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("hashchange", this.handleHashChange);
```
Comment (anujtekion) [inline src/commonComponents/sidebar/Sidebar.js:31]: please check if you can implement this using refs

File: src/components/dishItems/components/menu/Menu.js [hunk 24]
```diff
@@ -0,0 +1,70 @@
...
+      subtitle={currItem.length}
+      items={currItem}
+      renderRowItems={getMenuRowRenderProps}
+    />
+  );
+};
+
+const renderMenu = (items) => {
+  const itemCategories = _keys(items);
+  return _map(itemCategories, (itemCategory) =>
```
Comment (anujtekion) [inline src/components/dishItems/components/menu/Menu.js:39]: inline functions

**Pull request closed without merging**